pkg/controller/keyvault/ingress_secret_provider_class_test.go

```diff
@@ -136,6 +140,28 @@ func TestIngressSecretProviderClassReconcilerIntegration(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, testutils.GetErrMetricCount(t, ingressSecretProviderControllerName), beforeErrCount)
 	require.Greater(t, testutils.GetReconcileMetricCount(t, ingressSecretProviderControllerName, metrics.LabelSuccess), beforeRequestCount)
```

Comment:
we need tests guaranteeing that we aren't cleaning up spcs that we didn't create (don't have our labels)